backend/src/common/services/azure-ai/avatar-video/avatar-video-base.ts

```diff
@@ -0,0 +1,92 @@
+import { type Http } from 'shared';
+import { ApiPath, ContentType, HttpHeader, HTTPMethod } from 'shared';
+
+import { config } from '~/common/config/config.js';
+import { BaseHttpApi } from '~/common/http/http.js';
+import {
+    type GetAvatarVideoResponseDto,
+    type RenderAvatarVideoResponseDto,
+} from '~/common/services/azure-ai/types/types.js';
+
+import { AvatarApiPath } from './enums/enums.js';
+import { type RenderAvatarVideoArgument } from './types/types.js';
+
+type Constructor = {
+    baseUrl: string;
+    http: Http;
+};
+
+class AvatarVideoApi extends BaseHttpApi {
+    public constructor({ baseUrl, http }: Constructor) {
+        super({ path: ApiPath.AVATAR, baseUrl, http });
+    }
+
+    public async getAvatarVideo(
+        id: string,
+    ): Promise<GetAvatarVideoResponseDto> {
+        const response = await this.load(
+            this.getFullEndpoint(
+                `${AvatarApiPath.BATCHSYNTHESES}/${id}?api-version=2024-08-01`,
```

Comment:
Since `api-version` is a query param that is changeable it is better to move it to the constants especially if it is repeated in several places